src/Registration/cReg/NiftiImageData.cpp

```diff
@@ -183,6 +192,40 @@ std::shared_ptr<nifti_image> NiftiImageData<dataType>::create_from_geom_info(con
     return _nifti_image;
 }
 
+template<class dataType>
+void NiftiImageData<dataType>::construct_NiftiImageData_from_complex_im_real_component(std::shared_ptr<NiftiImageData> &out_sptr, const std::shared_ptr<const ImageData> in_sptr)
+{
+    // Create image from input
+    out_sptr = std::make_shared<NiftiImageData<dataType> >(*in_sptr);
+
+    ImageData::Iterator_const &it_in = in_sptr->begin();
+    typename NiftiImageData<dataType>::Iterator it_out = out_sptr->begin();
+    for (; it_in!=in_sptr->end(); ++it_in, ++it_out)
+        *it_out = (*it_in).complex_float().real();
+}
+
+template<class dataType>
+void NiftiImageData<dataType>::construct_NiftiImageData_from_complex_im_imag_component(std::shared_ptr<NiftiImageData> &out_sptr, const std::shared_ptr<const ImageData> in_sptr)
+{
+    if (!in_sptr->is_complex())
+        std::cout << "\nNiftiImageData<dataType>::construct_NiftiImageData_from_complex_im. Warning, input image is not complex. Complex component will be empty\n";
+
+    // Create image from input
+    out_sptr = std::make_shared<NiftiImageData<dataType> >(*in_sptr);
+
+    ImageData::Iterator_const &it_in = in_sptr->begin();
+    typename NiftiImageData<dataType>::Iterator it_out = out_sptr->begin();
+    for (; it_in!=in_sptr->end(); ++it_in, ++it_out)
+        *it_out = (*it_in).complex_float().imag();
```

Comment:
will this work if `*t_in` is real?

Reply:
Yes, think the num ref class is set up to be able to return a complex number from a real one. Just the imaginary component will be `0i`.

Reply:
ok